static/js/settings_streams.js

```diff
@@ -108,35 +141,17 @@ export function build_page() {
     meta.loaded = true;
 
     update_default_streams_table();
-
-    $(".create_default_stream").on("keypress", (e) => {
-        if (e.key === "Enter") {
-            e.preventDefault();
-            e.stopPropagation();
-            const default_stream_input = $(".create_default_stream");
-            make_stream_default(stream_data.get_stream_id(default_stream_input.val()));
-            default_stream_input[0].value = "";
-        }
-    });
-
-    $(".create_default_stream").typeahead({
-        items: 5,
-        fixed: true,
-        source() {
-            return stream_data.get_non_default_stream_names();
-        },
-        highlighter(item) {
-            return typeahead_helper.render_typeahead_item({primary: item});
-        },
-    });
+    init_dropdown_widgets();
 
     $(".default-stream-form").on("click", "#do_submit_stream", (e) => {
         e.preventDefault();
         e.stopPropagation();
-        const default_stream_input = $(".create_default_stream");
-        make_stream_default(stream_data.get_stream_id(default_stream_input.val()));
-        // Clear value inside input box
-        default_stream_input[0].value = "";
+        const default_stream_input = stream_data.maybe_get_stream_name(
+            Number.parseInt(default_stream_widget.value(), 10),
+        );
+        make_stream_default(stream_data.get_stream_id(default_stream_input));
```

Comment:
After line 152, I tried 
`console.log(stream_data.is_default_stream_id(stream_data.get_stream_id(default_stream_input)));`
It should print "true" as we are making the stream default, but it print "false", Due to this the streams list (in dropdown list widget) does not remove the stream (which is recently made default). How to fix it?